File: ru/codes/ruby/chapter_array_and_linkedlist/list.rb
=begin
File: list.rb
Created Time: 2024-03-18
Author: Xuan Khoa Tu Nguyen (ngxktuzkai2000@gmail.com)
=end

### Driver Code ###
if __FILE__ == $0
  # Инициализация списка
  nums = [1, 3, 2, 5, 4]
  puts "Список nums = #{nums}"

  # Доступ к элементу
  num = nums[1]
  puts "Элемент по индексу 1: num = #{num}"

  # Обновление элемента
  nums[1] = 0
  puts "После обновления элемента по индексу 1 до 0 nums = #{nums}"

  # Очистить список
  nums.clear
  puts "После очистки списка nums = #{nums}"

  # Добавление элемента в конец
  nums << 1
  nums << 3
  nums << 2
  nums << 5
  nums << 4
  puts "После добавления элементов nums = #{nums}"

  # Вставка элемента в середину
  nums.insert(3, 6)
  puts "После вставки элемента 6 по индексу 3 nums = #{nums}"

  # Удаление элемента
  nums.delete_at(3)
  puts "После удаления элемента по индексу 3 nums = #{nums}"

  # Обходить список по индексам
  count = 0
  for i in 0...nums.length
    count += nums[i]
  end

  # Непосредственно обходить элементы списка
  count = 0
  nums.each do |x|
    count += x
  end

  # Объединить два списка
  nums1 = [6, 8, 7, 10, 9]
  nums += nums1
  puts "После конкатенации списка nums1 к nums nums = #{nums}"

  nums = nums.sort { |a, b| a <=> b }
  puts "После сортировки списка nums = #{nums}"
end

File: vi/codes/typescript/chapter_array_and_linkedlist/list.ts
/**
 * File: list.ts
 * Created Time: 2022-12-10
 * Author: Justin (xiefahit@gmail.com)
 */

/* Initialize list */
const nums: number[] = [1, 3, 2, 5, 4];
console.log(`List nums = ${nums}`);

/* Update element */
const num: number = nums[1];
console.log(`Access element at index 1, get num = ${num}`);

/* Add elements at the end */
nums[1] = 0;
console.log(`Update element at index 1 to 0, get nums = ${nums}`);

/* Remove element */
nums.length = 0;
console.log(`After clearing list, nums = ${nums}`);

/* Direct traversal of list elements */
nums.push(1);
nums.push(3);
nums.push(2);
nums.push(5);
nums.push(4);
console.log(`After adding elements, nums = ${nums}`);

/* Sort list */
nums.splice(3, 0, 6);
console.log(`Insert number 6 at index 3, get nums = ${nums}`);

/* Remove element */
nums.splice(3, 1);
console.log(`Delete element at index 3, get nums = ${nums}`);

/* Traverse list by index */
let count = 0;
for (let i = 0; i < nums.length; i++) {
    count += nums[i];
}
/* Directly traverse list elements */
count = 0;
for (const x of nums) {
    count += x;
}

/* Concatenate two lists */
const nums1: number[] = [6, 8, 7, 10, 9];
nums.push(...nums1);
console.log(`After concatenating list nums1 to nums, get nums = ${nums}`);

/* Sort list */
nums.sort((a, b) => a - b);
console.log(`After sorting list, nums = ${nums}`);

export {};
